packages/metro/src/ModuleGraph/worker/collectDependencies.js

```diff
@@ -709,13 +732,36 @@ function createModuleNameLiteral(dependency: InternalDependency<mixed>) {
 class DefaultModuleDependencyRegistry<TSplitCondition = void>
   implements ModuleDependencyRegistry<TSplitCondition>
 {
+  /** Default key resolver. */
+  static getKeyForDependency(qualifier: ImportQualifier): string {
+    let key = qualifier.name;
+
+    // Add extra qualifiers when using `require.context` to prevent collisions.
+    if (qualifier.contextParams) {
+      // NOTE(EvanBacon): Keep this synchronized with `RequireContextParams`, if any other properties are added
+      // then this key algorithm should be updated to account for those properties.
+      // Example: `./directory__true__/foobar/m__lazy`
+      key += [
+        '',
```

Comment:
Let's future-proof slightly to allow us to distinguish between `require.context` and other things that may need extra qualifiers.

```suggestion
        '',
        'context',
```